### ISSUE TITLE
Fine tune the network?

### ISSUE THREAD

**Issue by sbeesm** [matterport/Mask_RCNN#1960]
I want to train the model on custom data but the dataset size is small.
Does this implementation provide a method to freeze the initial layers and train the model only on some last layers?

**Comment by aashokvardhan** [comment]
The train method gives you the flixibility of how many layers you want to train.Code snippets how describes the functionality.

mrcnn/model.py:

layers: Allows selecting which layers to train. It can be:
            - A regular expression to match layer names to train
            - One of these predefined values:
              heads: The RPN, classifier and mask heads of the network
              all: All the layers
              3+: Train Resnet stage 3 and up
              4+: Train Resnet stage 4 and up
              5+: Train Resnet stage 5 and up

samples/nucleus/nucleus.py:

 print("Train network heads")
    model.train(dataset_train, dataset_val,
                learning_rate=config.LEARNING_RATE,
                epochs=20,
                augmentation=augmentation,
                layers='heads')

**Comment by sbeesm** [comment]
Thank you for the response @aashokvardhan, this solves the issue.
